content/blog/2022/2022-06-30-configuring-cors-with-spring.md

```diff
@@ -0,0 +1,430 @@
+
+````java
+    @CrossOrigin(origins = "http://localhost:4200", allowedHeaders = "Requestor-Type", exposedHeaders = "X-Get-Header")
+    @GetMapping
+    public ResponseEntity<Mono<List<BookDto>>> getBooks(@RequestParam String type) {
+        HttpHeaders headers = new HttpHeaders();
+        headers.set("X-Get-Header", "ExampleHeader");
+        return ResponseEntity.ok().headers(headers).body(libraryService.getAllBooks(type));
+    }
+````
+
+#### Enable CORS Configuration globally in Spring Webflux
+
+To define CORS globally in a Spring Webflux application, we use the **WebfluxConfigurer** and override the **addCorsMappings()**.
+Similar to Spring REST, it uses a **CorsConfiguration** with defaults that can be overridden as required.
+
+````java
+    @Bean
+    public WebFluxConfigurer corsMappingConfigurer() {
+            return new WebFluxConfigurer() {
+        @Override
+        public void addCorsMappings(CorsRegistry registry) {
+            WebConfigProperties.Cors cors = webConfigProperties.getCors();
+            registry.addMapping("/**")
+            .allowedOrigins(cors.getAllowedOrigins())
+            .allowedMethods(cors.getAllowedMethods())
+            .maxAge(cors.getMaxAge())
+            .allowedHeaders(cors.getAllowedHeaders())
+            .exposedHeaders(cors.getExposedHeaders());
+            }
+        };
+    }
+````
+
+{{% info title="NOTE:" %}}
+In all the above cases we can define both global CORS configuration and local configuration using @CrossOrigin.
+For attributes that accept multiple values, a combination of global and local values will apply. For attributes that accept
+only a single value, the local value will take precedence over the global one.
+{{% /info %}}
+
+#### Enable CORS using WebFilter
+
+The Webflux framework allows CORS configuration to be set globally via **CorsWebFilter**. We can use the **CorsConfiguration** object to set 
+the required configuration and register **CorsConfigurationSource** to be used with the filter. However, by default the CorsConfiguration in case of filters
+does not assign default configuration to the endpoints. Only the specified configuration can be applied.
+Another option is to call **CorsConfiguration.applyPermitDefaultValues()** explicitly.
+
+````java
+        @Bean
+        public CorsWebFilter corsWebFilter() {
+            CorsConfiguration corsConfig = new CorsConfiguration();
+            corsConfig.setAllowedOrigins(Arrays.asList("http://localhost:4200"));
+            corsConfig.setMaxAge(3600L);
+            corsConfig.addAllowedMethod("*");
+            corsConfig.addAllowedHeader("Requestor-Type");
+            corsConfig.addExposedHeader("X-Get-Header");
+
+            UrlBasedCorsConfigurationSource source =
+                new UrlBasedCorsConfigurationSource();
+            source.registerCorsConfiguration("/**", corsConfig);
+
+            return new CorsWebFilter(source);
+    }
+````
+
+## Enabling CORS with Spring Security
```

Comment:
The headers are a bit inconsistent. Some use the "ing" form like "Enabling ...", and some use the form "Enable". Can you please go over them and make them consistent?